mesop/server/server.py

```diff
@@ -126,15 +145,34 @@ def generate_data(ui_request: pb.UiRequest) -> Generator[str, None, None]:
         runtime().context().reset_current_node()
 
         result = runtime().context().run_event_handler(ui_request.user_event)
+        has_run_navigate_on_load = False
         for _ in result:
           path = ui_request.path
           for command in runtime().context().commands():
             if command.HasField("navigate"):
               path = command.navigate.url
+              page_config = runtime().get_page_config(path=path)
```

Comment:
[nit] I feel like this section, lines 154-170 is could be cleaned up a bit code-wise. No real suggestion though. It's not entirely the same as the on load behavior above. But seems very similar. I think ok for now since we don't want to make a poor abstraction.

I think the one part I have trouble understanding is the use of `has_run_navigate_on_load`. What happens on when there are multiple navigate commands? Does that mean we're navigating to multiple different pages? If so, seems like we'd need to run the load event each time if each page had an onload?

Reply:
So to clarify, you can do this:

```py
def on_load():
  me.navigate("/p1")
  yield
  me.navigate("/p2")
  yield
```

This probably isn't a great UX, but it's allow-able and p1's onload will be executed on the first tick and then p2's onload will be executed on the second tick (note: p1 is cleared from the commands after it's been run once).

- Added a warning if there's multiple navigate commands found at once.